### DISCUSSION TITLE
How can I reconnect my console command after disconnection?

### DISCUSSION THREAD

**Discussion by Kimmydoray** [spatie/laravel-twitter-streaming-api#51] [Q&A]
I am running the twitter streaming api in my background using laravel console command, but once I disconnect my internet connection it shows this error: Unable to read from stream(please see attached image)
![image](https://user-images.githubusercontent.com/44127458/210777671-1352dddf-91bd-44e6-8023-34468bcc8b54.png)

How can I reconnect it?

**Comment by felixdorn** [comment]
You need internet access to read from the stream. If you want to recover lost messages during a time period, you can only do so if you have "Academic Research / Entreprise" access. Even if you do have one of those tier, you can only backfill the last 5 minutes. You can also use the Recovery feature to recover missed data within the last 24 hours if you are unable to reconnect with the 5 minute backfill window, however that's using a different endpoint not supported by any of the PHP packages out there.

To reconnect it, simply re-run the command, or handle the exception in your code that re-starts the listener.

You may use the following boilerplate to make it work easily:

```php
public function startWorking() {
     try {
          // start the stream
    } catch (RuntimeException $e) {
          report($e);
          
          if (/* exception is not "unable to read stream" */) {
               return;
          }

          sleep(5)

          $this->startWorking();
    }
}
```

You may want to add some system to stop retrying after a number of times. I do not recommend using exponential back-off here as it would make little sense.
